test/hack/nodeport-ip-aws/setup.sh

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# Adjust to point to your AWS credentials file
+AWSCREDS="${AWSCREDS:-${HOME}/.aws/credentials}"
+
+# Adjust to point to your desired SSH public and private key file
+SSHPUBLICKEY="${SSHPUBLICKEY:-${HOME}/.ssh/id_rsa.pub}"
+SSHPRIVATEKEY="${SSHPRIVATEKEY:-${HOME}/.ssh/id_rsa}"
+
+# Adjust to point to the right hypershift binary
+HYPERSHIFT="${HYPERSHIFT:-./bin/hypershift}"
+
+# Obtain the infra-id and region from the current management cluster
+INFRAID="$(oc get infrastructure/cluster -o jsonpath='{ .status.infrastructureName }')"
+REGION="$(oc get infrastructure/cluster -o jsonpath='{ .status.platformStatus.aws.region }')"
+
+export AWS_SHARED_CREDENTIALS_FILE="${AWSCREDS}"
+
+# Create a bastion machine for the management cluster
+${HYPERSHIFT} create bastion aws \
+   --aws-creds "${AWSCREDS}" \
+   --infra-id "${INFRAID}" \
+   --region "${REGION}" \
+   --ssh-key-file "${SSHPUBLICKEY}"
+
+# Query a description of the bastion machine
+INSTANCEFILE="$(mktemp)"
+aws ec2 describe-instances --filters "Name=tag:Name,Values=${INFRAID}-bastion" > "${INSTANCEFILE}"
+
+# Obtain the machine's public IP
+PUBLICIP="$(cat "${INSTANCEFILE}" | jq -r '.Reservations[].Instances[] | select(.State.Name == "running") | .PublicIpAddress')"
+
+# Obtain the machine's security group
+SGID="$(cat "${INSTANCEFILE}" | jq -r '.Reservations[].Instances[] | select(.State.Name == "running") | .SecurityGroups[0].GroupId')"
+
+# Add permission for service port range to security group
+aws ec2 authorize-security-group-ingress \
+   --group-id "${SGID}" \
+   --protocol tcp \
+   --port 32000-32767 \
+   --cidr "0.0.0.0/0"
+
+# Find the master machines security group
+MASTERSGID="$(aws ec2 describe-security-groups --filters "Name=tag:Name,Values=${INFRAID}-master-sg" | jq -r '.SecurityGroups[0].GroupId')"
+
+# Allow the master machines access from the bastion sg
+aws ec2 authorize-security-group-ingress \
+   --group-id "${MASTERSGID}" \
+   --protocol tcp \
+   --port 32000-32767 \
+   --source-group "${SGID}"
+
+# Find IP of a master machine
+MASTERIP="$(aws ec2 describe-instances --filters Name=tag:Name,Values=${INFRAID}-master-0 | jq -r '.Reservations[].Instances[0].PrivateIpAddress')"
+
+# Enable IP forwarding on bastion machine
+ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -i "${SSHPRIVATEKEY}" "ec2-user@${PUBLICIP}" \
```

Comment:
Can we not do that? How a private key is provided is IMHO an env config that is outside of the scope of this script. The main issue with the current approach is that it can not work for encrypted ssh keys that are provided through a ssh agent and always require configs if the ssh key is not in `${HOME}/.ssh/id_rsa` (which is something the `ssh` cli defaults to anyways)

Reply:
Sure, will remove the need for a private key and update the instructions to do an ssh-add if needed.